### PR TITLE
Add relay.where

### PR DESCRIPTION
**Pull request by zhiics**
Add the `where` operator. BTW, I think **where** is not good name for this operator. It actually performs similarly what `blend` instruction does in x86 (https://www.felixcloutier.com/x86/BLENDVPS.html).

CC @tqchen @yzhliu

### CODE REVIEW

**Comment by junrushao1994** [comment]
I agree that the name `where` sucks, but MATLAB and numpy seem to actually use that name since the beginning of time. If our target customers are machine learning engineers, it could be great because that is the way they think of things.

An absolutely perfect solution would be that we could introduce aliases of this operator, like `blend`, if you are interested.

**Comment by zhiics** [comment]
@junrushao1994 Thanks for the information. I will just keep the name for now.

File: docs/langref/relay_op.rst [hunk 1]
```diff
@@ -169,3 +170,4 @@ Level 4 Definitions
 Level 5 Definitions
 -------------------
 .. autofunction:: tvm.relay.image.resize
+.. autofunction:: tvm.relay.where
```
Comment (yzhliu) [inline docs/langref/relay_op.rst]: It should be in Level 4?

File: python/tvm/relay/op/transform.py [hunk 2]
```diff
@@ -139,3 +139,40 @@ def take(data, indices, axis=None):
...
+
+
+def where(condition, x, y):
+    """Selecting elements from either x or y depending on the value of the
+    condition.
+
+    Parameters
+    ----------
+    condition : relay.Expr
+        The condition array.
```
Comment (yzhliu) [inline python/tvm/relay/op/transform.py]: add more details, e.g., when 0 yield y, otherwise yield x.

File: python/tvm/relay/op/transform.py [hunk 3]
```diff
@@ -139,3 +139,40 @@ def take(data, indices, axis=None):
...
+    --------
+    .. code-block:: python
+
+        x = [[1, 2], [3, 4]]
+        y = [[5, 6], [7, 8]]
+        condition = [[0, 1], [-1, 0]]
+        where(conditon, x, y) = [[5, 2], [3, 8]]
+
+        condition = [1, 0]
+        where(conditon, x, y) = [[5, 2], [3, 8]]
```
Comment (yzhliu) [inline python/tvm/relay/op/transform.py]: is it true?

Comment (zhiics) [reply python/tvm/relay/op/transform.py]: @yzhliu oops. My bad, forgot to update the values.

**Comment by MarisaKirisame** [comment]
I am very supportive of @junrushao1994 's idea, and I think we should add another name for where. Any thought?

**Comment by zhiics** [comment]
@junrushao1994 @MarisaKirisame Thanks for the thoughts. TMHO, it looks a little strange to me to have "two" operators doing the same thing. Among these two options, I think providing an alias frontend API would be more suitable since the users/customers mostly only take APIs. But I am open for any other inputs.

**Comment by junrushao1994** [comment]
@tqchen could you comment on the alias thing?

**Comment by tqchen** [comment]
To keep things simple, let us just be consistent with numpy and do not introduce alias for now

File: python/tvm/relay/op/transform.py [hunk 4]
```diff
@@ -139,3 +139,42 @@ def take(data, indices, axis=None):
...
+
+        x = [[1, 2], [3, 4]]
+        y = [[5, 6], [7, 8]]
+        condition = [[0, 1], [-1, 0]]
+        relay.where(conditon, x, y) = [[5, 2], [3, 8]]
+
+        condition = [1, 0]
+        relay.where(conditon, x, y) = [[1, 2], [7, 8]]
+
+    Note that the shape of condition, x, and y needs to be consistent.
```
Comment (junrushao1994) [inline python/tvm/relay/op/transform.py]: Could we make it more clear what “consistent” means? e.g. who is broadcastable from whom

Comment (zhiics) [reply python/tvm/relay/op/transform.py]: I will just say the "same of condition, x, and y needs to be the same"

Comment (junrushao1994) [reply python/tvm/relay/op/transform.py]: @zhiics sounds cool

**Comment by zhiics** [comment]
@tqchen @junrushao1994 Okay, let's then just keep it like this. I will rebase and commit again. Thanks.

**Comment by zhiics** [comment]
@tqchen I think it is ready to land, could you please take a look? Thanks.

**Pull request merged**